src/core/org/apache/jmeter/services/FileServer.java

```diff
@@ -418,16 +418,20 @@ private BufferedReader getReader(String alias, boolean recycle, boolean firstLin
     }
 
     private BufferedReader createBufferedReader(FileEntry fileEntry) throws IOException {
-        FileInputStream fis = new FileInputStream(fileEntry.file);
-        InputStreamReader isr = null;
-        // If file encoding is specified, read using that encoding, otherwise use default platform encoding
-        String charsetName = fileEntry.charSetEncoding;
-        if(!JOrphanUtils.isBlank(charsetName)) {
-            isr = new InputStreamReader(fis, charsetName);
+        if (!fileEntry.file.exists() || !fileEntry.file.canRead() || !fileEntry.file.isFile()) {
+            throw new IllegalArgumentException("File "+ fileEntry.file.getName()+ " must exist and be readable");
         } else {
```

Comment:
Please remove `else` and the following braces.
It would avoid indenting the following code.